File: solutions/golang/bestTimeToBuyAndSellStockII/solution_test.go
package main

import "testing"

func TestMaxProfit(t *testing.T) {
  tests := []struct {
    prices []int
    ans int
  } {
    {
      prices: []int{7, 1, 5, 3, 6, 4},
      ans: 7,
    },
    {
      prices: []int{1, 2, 3, 4, 5},
      ans: 4,
    },
    {
      prices: []int{7, 6, 4, 3, 1},
      ans: 0,
    },
  }
  for _, tt := range tests {
    actual := maxProfit(tt.prices)
    if tt.ans != actual {
      t.Errorf("got %d for input %v; expected %d",
        actual, tt.prices, tt.ans)
    }
  }
}

File: solutions/cpp/courseSchedule/solution_test.cpp
//
// Created by mingyi on 07.06.20.
//


#include "gtest/gtest.h"
#include "gmock/gmock.h"

#include "solution.cpp"
#include "../include/structures.hpp"

using namespace std;
using namespace testing;

typedef TestStruct<tuple<int, vector<vector<int>>>, bool> TStruct;

class CanFinishTest : public ::testing::TestWithParam<TStruct> {};

TEST_P(CanFinishTest, Test
) {
  TStruct tt = GetParam();
  auto[numCourses, prerequisites] = tt.input;
  bool actual = canFinish(numCourses, prerequisites);
  ASSERT_EQ(tt.expected, actual);
}

INSTANTIATE_TEST_SUITE_P(Tests, CanFinishTest, ::testing::Values(
    TStruct{
        .input = make_tuple<int, vector<vector<int>>>(2, {{1, 0}}),
        .expected = true
    },
    TStruct{
        .input = make_tuple<int, vector<vector<int>>>(2, {{1, 0}, {0, 1}}),
        .expected = false
    }
)
);